src/main/java/io/cryostat/net/web/http/api/v1/TargetSnapshotPostHandler.java

```diff
@@ -120,7 +131,40 @@ public void handleAuthenticated(RoutingContext ctx) throws Exception {
 
                             return rename;
                         });
-        ctx.response().setStatusCode(200);
-        ctx.response().end(result);
+
+        Optional<InputStream> snapshotOptional =
+                recordingTargetHelper.getRecording(connectionDescriptor, result).get();
+        if (snapshotOptional.isEmpty()) {
+            throw new HttpStatusException(
+                    500,
+                    String.format(
+                            "Successful creation verification of snapshot %s failed", result));
+        } else if (snapshotIsEmpty(snapshotOptional.get())) {
+            recordingTargetHelper.deleteRecording(connectionDescriptor, result).get();
+            ctx.response().setStatusCode(202);
+            ctx.response()
+                    .setStatusMessage(
+                            "Snapshot failed to create: Cryostat is not aware of any Active, non-Snapshot source recordings to take event data from");
+            ctx.response().end();
+        } else {
+            ctx.response().setStatusCode(200);
+            ctx.response().end(result);
+        }
+    }
+
+    private boolean snapshotIsEmpty(InputStream snapshot) throws IOException {
+        PushbackInputStream pushbackSnapshot = new PushbackInputStream(snapshot);
+        try {
+            int b = pushbackSnapshot.read();
+            // If this point is reached (i.e. no IOException was thrown) the stream must
+            // be non-empty so push back the last read byte
+            pushbackSnapshot.unread(b);
+            return false;
+        } catch (IOException e) {
+            if (e.getMessage().equals("java.io.IOException: No recording data available")) {
```

Comment:
Relying on the string contents here looks fragile. Is there any other way to perform this check?

Reply:
From my initial debugging, the message string was what alerted me to the fact that an `IOException` is thrown when attempting to read empty streams. I can take another look and see if there are any other identifying fields in the `IOException` object.

Reply:
Just realized, by "this check" do you mean checking if the `IOException` is due to an empty stream or if there exists another method for checking if a stream is empty?

Reply:
It looks like this is roughly the code that throws that exception:

https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/master/src/jdk.management.jfr/share/classes/jdk/management/jfr/FlightRecorderMXBeanImpl.java#L150

This may change with different target JVM versions that we're communicating with, which is why I say that the check seems fragile. I don't see any immediately obvious other way to distinguish this kind of IOException from any others, unfortunately. Perhaps this method should be renamed from `snapshotIsEmpty` to `snapshotIsReadable` or something, and if any IOException is thrown when trying to read the first byte for any reason, then it's `false`. If the snapshot is unreadable immediately after creating it then perhaps the most likely reason is that it's simply empty, but it may be ambiguous. In any case I suppose we want to delete the unreadable snapshot.

Reply:
Ah, that makes sense, right. I agree with the `snapshotIsReadable` approach. I can change the HTTP API to say that the 202 response is due to "an unreadable snapshot, likely caused by a lack of Active source recordings..." (something along those lines). I'll do that for front-end notification as well.